ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/ListIndexTool.java

```diff
@@ -0,0 +1,576 @@
+    public static final String TYPE = "ListIndexTool";
+    // This needs to be changed once it's changed in opensearch core in RestIndicesListAction.
+    private static final int MAX_SUPPORTED_LIST_INDICES_PAGE_SIZE = 5000;
+    public static final int DEFAULT_PAGE_SIZE = 100;
+    private static final String DEFAULT_DESCRIPTION = String
+        .join(
+            " ",
+            "This tool gets index information from the OpenSearch cluster.",
+            "It takes 2 optional arguments named `index` which is a comma-delimited list of one or more indices to get information from (default is an empty list meaning all indices),",
+            "and `local` which means whether to return information from the local node only instead of the cluster manager node (default is false).",
+            "The tool returns the indices information, including `health`, `status`, `index`, `uuid`, `pri`, `rep`, `docs.count`, `docs.deleted`, `store.size`, `pri.store. size `, `pri.store.size`, `pri.store`."
+        );
+
+    @Setter
+    @Getter
+    private String name = ListIndexTool.TYPE;
+    @Getter
+    @Setter
+    private String description = DEFAULT_DESCRIPTION;
+    @Getter
+    private String version;
+
+    private Client client;
+    @Setter
+    private Parser<?, ?> inputParser;
+    @Setter
+    private Parser<?, ?> outputParser;
+    @SuppressWarnings("unused")
+    private ClusterService clusterService;
+
+    public ListIndexTool(Client client, ClusterService clusterService) {
+        this.client = client;
+        this.clusterService = clusterService;
+
+        outputParser = new Parser<>() {
+            @Override
+            public Object parse(Object o) {
+                @SuppressWarnings("unchecked")
+                List<ModelTensors> mlModelOutputs = (List<ModelTensors>) o;
+                return mlModelOutputs.get(0).getMlModelTensors().get(0).getDataAsMap().get("response");
+            }
+        };
+    }
+
+    @Override
+    public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
+        // TODO: This logic exactly matches the OpenSearch _list/indices REST action. If code at
+        // o.o.rest/action/list/RestIndicesListAction.java changes those changes need to be reflected here
+        @SuppressWarnings("unchecked")
+        List<String> indexList = parameters.containsKey("indices")
+            ? gson.fromJson(parameters.get("indices"), List.class)
+            : Collections.emptyList();
+        final String[] indices = indexList.toArray(Strings.EMPTY_ARRAY);
+
+        final IndicesOptions indicesOptions = IndicesOptions.strictExpand();
+        final boolean local = parameters.containsKey("local") && Boolean.parseBoolean(parameters.get("local"));
+        final boolean includeUnloadedSegments = Boolean.parseBoolean(parameters.get("include_unloaded_segments"));
+        final PageParams pageParams = new PageParams(null, PageParams.PARAM_ASC_SORT_VALUE, DEFAULT_PAGE_SIZE);
```

Comment:
can we take page size as a parameter input like:

```
final int pageSize = parameters.containsKey("page_size")
            ? Integer.parseInt(parameters.get("page_size"))
            : DEFAULT_PAGE_SIZE;

Reply:
Done